janitor/functions/change_type.py

```diff
@@ -62,15 +65,21 @@ def change_type(
     elif ignore_exception == "keep_values":
         df[column_name] = df[column_name].astype(dtype, errors="ignore")
     elif ignore_exception == "fillna":
+        if isinstance(column_name, Hashable):
+            column_name = [column_name]
 
-        def convert(x, dtype):
-            """Casts item `x` to `dtype` or None if not possible."""
-            try:
-                return dtype(x)
-            except ValueError:
-                return None
-
-        df[column_name] = df[column_name].apply(lambda x: convert(x, dtype))
+        df[column_name] = df[column_name].apply(
+            lambda s: s.map(lambda x: _convert(x, dtype))
+        )
```

Comment:
A little patch to `fillna`. Since we use `dtype(x)` logic. This only can handle a single element at once.
So we need to iterate the series, then iterate its values.

Reply:
refactor via `.applymap` to convert all values